test/unit/math/binaryOps/src/binaryOps.cpp

```diff
@@ -0,0 +1,432 @@
+            switch (structs[gridThreadIdx].range)
+            {
+                case Range::POSITIVE_ONLY:
+                    for(TIdx i(0); i < sizeArgs/2 -1; ++i)
+                    {
+                        argX = argsX[i];
+                        argY = argsY[i];
+                        results[i + gridThreadIdx* sizeArgs] = structs[gridThreadIdx].alpakaOp(acc, argX, argY);
+                    }
+                    break;
+                case Range::NOT_ZER0:
+                    for(TIdx i(0); i < sizeArgs; ++i)
+                    {
+                        if(i == sizeArgs/2 -1 || i == sizeArgs/2)
+                            continue;
+                        argX = argsX[i];
+                        argY = argsY[i];
+                        results[i + gridThreadIdx* sizeArgs] = structs[gridThreadIdx].alpakaOp(acc, argX, argY);
+                    }
+                    break;
+
+                case Range::X_NOT_ZERO:
+                    for(TIdx i(0); i < sizeArgs/2 -1; ++i)
+                    {
+                        if(i == sizeArgs/2 -1 || i == sizeArgs/2)
+                            argX = argsX[0];
+                        else
+                            argX = argsX[i];
+                        argY = argsY[i];
+                        results[i + gridThreadIdx* sizeArgs] = structs[gridThreadIdx].alpakaOp(acc, argX, argY);
+                    }
+                    break;
+
+                case Range::Y_NOT_ZERO:
+                    for(TIdx i(0); i < sizeArgs/2 -1; ++i)
+                    {
+                        if(i == sizeArgs/2 -1 || i == sizeArgs/2)
+                            argY = argsY[0];
+                        else
+                            argY = argsY[i];
+                        argX = argsX[i];
+
+                        results[i + gridThreadIdx* sizeArgs] = structs[gridThreadIdx].alpakaOp(acc, argX, argY);
+                    }
+                    break;
+
+                case Range::UNRESTRICTED:
+                    for(TIdx i(0); i < sizeArgs; ++i)
+                    {
+                        argX = argsX[i];
+                        argY= argsY[i];
+                        results[i + gridThreadIdx* sizeArgs] = structs[gridThreadIdx].alpakaOp(acc, argX, argY);
+                    }
+                    break;
+
+                default:
+                    break;
+            }
+        }
+    }
+};
+
+
+struct TestTemplate {
+    template<typename TAcc>
+    void operator()() {
+        using Dim = alpaka::dim::Dim<TAcc>;
+        using Idx = alpaka::idx::Idx<TAcc>;
+        using Data = double;
+        using DevAcc = alpaka::dev::Dev<TAcc>;
+        using PltfAcc = alpaka::pltf::Pltf<DevAcc>;
+        using QueueAcc = alpaka::test::queue::DefaultQueue<DevAcc>;
+        using PltfHost = alpaka::pltf::PltfCpu;
+
+
+        std::cout << "\nTesting next AccType \n\n";
+        // the functions that will be tested
+
+
+        TestStruct<TAcc, Data> arr[] =
```

Comment:
could you please use `std:Array` than we can later in the code use `array.size()` instead `Idx const numOps = sizeof(arr) / sizeof(TestStruct<TAcc, Data>);`

Reply:
Yes but than we have to specialize the size in the initialization, because there is no easy conversion from `T arr [] = {...}` to `std::array<T, ?> arr = {...}`

Reply:
Unfortunately so. Only this [experimental one](https://en.cppreference.com/w/cpp/experimental/make_array) (I do not suggest to use it here).